src/Relude/Extra/Foldable1.hs

```diff
@@ -189,6 +198,66 @@ instance (Foldable1 f, Foldable1 g) => Foldable1 (Sum f g) where
     foldMap1 f (InR y) = foldMap1 f y
     {-# INLINE foldMap1 #-}
 
+
+----------------------------------------------------------------------------
+-- List custom error
+----------------------------------------------------------------------------
+
+-- | For tracking usage of ordinary list with 'Foldable1' functions.
+type family IsListError :: Constraint
+  where
+    IsListError = TypeError
+        ( 'Text "The methods of the 'Foldable1' type class work with non-empty containers."
+        ':$$: 'Text "However, one of the 'Foldable1' functions is applied to the List."
+        ':$$: 'Text ""
+        ':$$: 'Text "Possible fix:"
+        ':$$: 'Text "  * Replace []"
+        ':$$: 'Text "    with one of the: 'NonEmpty', 'Identity', '(c,)', 'Conpose f g', 'Product f g', 'Sum f g'"
+        ':$$: 'Text "  * Or use 'Foldable' class for your own risk."
+        )
+
+
+{- | __CAUTION:__ This instance is for custom error display only.
```

Comment:
Oooh, I like this __CAUTION__ idea!! It improves the quality of error messages by a lot. Should we also go even further and use emojis? Like 

:warning: __CAUTION__ :warning:

Reply:
I LOVE IT! But would it work on Windows?

Reply:
Well, it's for Haddock only, so it will work. But if we put it into the error itself, I'm not sure... I never saw custom type errors with emojis...

Reply:
Me too, could possibly be not accepted gratefully by users, I'm afraid...